### PR TITLE
Convert to TS

### PR DESCRIPTION
**Pull request by davidmeirlevy**
issue #2

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  return proxy({
+    target,
+    changeOrigin: true,
+    headers: {
+      tenant,
+    },
+  });
+}
+
+function getProxyTarget(service: IServiceProxyConfig) {
```
Comment (or-yam) [inline src/index.ts:22]: Why not arrow functions ?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  const {
+    authService,
+    contentService,
+    adminPanel,
+    assetsService,
+    draftsService,
+    tenant,
+    applicationUrl,
+    excludedServices
+  } = { ...getApiProxyConfig(), ...config }
```
Comment (or-yam) [inline src/index.ts:36]: Why prefer spreading over `Object.assign` ?

File: src/index.ts (hunk 1 above)
Comment (davidmeirlevy) [reply src/index.ts:22]: Easier to define types.

File: src/index.ts (hunk 2 above)
Comment (davidmeirlevy) [reply src/index.ts:36]: I always prefer syntax operation over functional.

**Review by or-yam — APPROVED** [review]

**Review by drorrose — APPROVED** [review]

**Pull request merged**